includes/WikiPage.php

```diff
@@ -2853,20 +2853,25 @@ public function updateCategoryCounts( $added, $deleted ) {
 		// Wikia change - begin - @author: wladek
 		// PLATFORM-410: Attempt to lower the chance of deadlocks
 		sort( $insertCats );
+		$missingCats = [];
 		foreach ( $insertCats as $cat ) {
-			$dbw->selectRow( 'category', 'cat_id', [ 'cat_title' => $cat ], __METHOD__, [ 'FOR UPDATE' ] );
+			if ( !$dbw->selectRow( 'category', 'cat_id', [ 'cat_title' => $cat ], __METHOD__, [ 'FOR UPDATE' ] ) ) {
+				$missingCats[] = $cat;
+			}
 		}
-		// Wikia change - end
+		$insertCats = $missingCats;
 
-		$insertRows = array();
-
-		foreach ( $insertCats as $cat ) {
-			$insertRows[] = array(
-				'cat_id' => $dbw->nextSequenceValue( 'category_cat_id_seq' ),
-				'cat_title' => $cat
-			);
+		if ( $insertCats ) {
```

Comment:
moving this "if" check closer to the insert() function makes it a bit easier to scan